### PR TITLE
feat: notification action routing

### PR DESCRIPTION
**Pull request by SHAH-JI**
Description:

- You can send the path in the notification payload and when user receive the notification, the app will navigate according to path
Example:
```dart
createNotification(title: 'Test Notification', body: 'This is a test notification', payload: {'path': '/chat'});
```
- For more screens in the future, you  can update the `screensWithRespectToPath` variable in `notification.dart` file

Changes:

- Added Navigator key for navigation in the MyApp in main.dart
- Added a setter and getter for the screen position in SharedPreferencesUtil

### CODE REVIEW

**Comment by SHAH-JI** [comment]
to @josancamon19 Please check the PR and if there any changes requested let me know.
Thanks.

**Review by josancamon19 — CHANGES_REQUESTED** [review]
Please use 120 chars line length for linting.

**Comment by mshahjahan-mufin** [comment]
to @josancamon19 Please confirm, I have set it to 120 chars line length.

**Comment by josancamon19** [comment]
Taking a look at it rn

**Comment by josancamon19** [comment]
Hi @mshahjahan-mufin, weird, I still see linting issues:

Use android studio linting with the rule change of 120 chars.
Or you can use `dart format`
CleanShot 2024-06-28 at 17 20 43@2x

**Comment by josancamon19** [comment]
This implementation looks great, what could we do if we want to access, a specific memory or settings or plugins from the notification? Let's implement this @mshahjahan-mufin

**Comment by josancamon19** [comment]
We can do some other parameter in payload like navigateTo, and handle that on home/page.dart.

Still allowing to appear on one of the main pages as desired, but also to simulate like a forward click to another page.

File: apps/AppWithWearable/lib/pages/home/page.dart [hunk 1]
```diff
@@ -97,7 +97,12 @@ class _HomePageWrapperState extends State<HomePageWrapper> with WidgetsBindingOb
...
   @override
   void initState() {
-    _controller = TabController(length: 3, vsync: this, initialIndex: 1);
+    _controller = TabController(
+      length: 3,
+      vsync: this,
+      initialIndex: SharedPreferencesUtil().pageToShowFromNotification,
+    );
+    SharedPreferencesUtil().pageToShowFromNotification = 1;
     SharedPreferencesUtil().onboardingCompleted = true;
```
Comment (josancamon19) [inline apps/AppWithWearable/lib/pages/home/page.dart:113]: Also, please include payload /chat to this notification:

```
    createNotification(
      title: 'Here is your action plan for tomorrow',
      body: 'Check out your daily summary to see what you should do tomorrow.',
      notificationId: 5,
      isDailySummaryNotification: true,
      payload: {'path': '/chat'},
    );
```

**Review by josancamon19 — CHANGES_REQUESTED** [review]

**Comment by SHAH-JI** [comment]
to @josancamon19 
Changes done.
1. Character line set 120 and formatted.
2. Subpage routing completed, you can send the subpage route into `payload['navigateTo']` and add the screen in the `screensWithRespectToPath` JSON object in `home/page.dart` file.

**Review by josancamon19 — APPROVED** [review]

**Comment by josancamon19** [comment]
@SHAH-JI Great job, thanks! 🎉

Please send me your transfer details to joan.santiago.cabezas@gmail.com

**Pull request merged**